server/aesdsocket.c

```diff
@@ -110,56 +130,98 @@ void listen_for_connections(int sockfd) {
 }
 
 // Handles client connections
-void handle_client(int client_fd) {
-    FILE *fp = fopen(CUSTOM_LOG_FILE, "a+");
+void *handle_client(void *arg) {
+    int client_fd = *((int *)arg);
+    free(arg);
+    
+    add_thread(pthread_self());
+    
     char buffer[MAX_CUSTOM_BUFFER];
+    char packetBuffer[MAX_CUSTOM_BUFFER];  // Buffer to accumulate a complete packet
+    size_t packetSize = 0;
 
     while (1) {
         ssize_t bytes_recv = recv(client_fd, buffer, sizeof(buffer), 0);
         if (bytes_recv <= 0) {
             break;
         }
 
-        fwrite(buffer, bytes_recv, 1, fp);
+        // Accumulate data until a complete packet is received
+        memcpy(packetBuffer + packetSize, buffer, bytes_recv);
+        packetSize += bytes_recv;
 
+        // Check if a complete packet (ending with '\n') is received
         if (memchr(buffer, '\n', bytes_recv) != NULL) {
+            // Write the complete packet to the log file using system call
+            pthread_mutex_lock(&list_mutex);
+            int fd = open(CUSTOM_LOG_FILE, O_RDWR | O_CREAT | O_APPEND, 0777);
+            if (fd == -1) {
+                syslog(LOG_INFO, "Couldn't open file");
+                pthread_exit(NULL);
```

Comment:
You are not supposed to exit thread. pthread_join should take care from main() context